### PR TITLE
API for signIn case added. View connected to the backEnd

### PR DESCRIPTION
**Pull request by pablodri08**
* [Connecting SignIn View to the backEnd](https://androidtraining.atlassian.net/browse/ATP-166?atlOrigin=eyJpIjoiMGRjNTg3NGZhZmU5NDlkZTg0NjQyYjg5Yjc5YzI4ZjAiLCJwIjoiaiJ9)

resized_medium

### CODE REVIEW

File: app/src/main/java/com/xmartlabs/taskloans/data/service/OutServiceApi.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.xmartlabs.taskloans.data.service
+
+import com.xmartlabs.taskloans.data.model.User
+import com.xmartlabs.taskloans.data.model.service.SignInRequest
+import com.xmartlabs.taskloans.data.model.service.SignInResponse
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface OutServiceApi {
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/taskloans/data/service/OutServiceApi.kt]: Use a AuthServiceApi

File: app/src/main/java/com/xmartlabs/taskloans/ui/screens/signin/SignInFragment.kt [hunk 2]
```diff
@@ -30,16 +34,18 @@ class SignInFragment : BaseViewBindingFragment<FragmentSigninBinding>() {
...
-            ).show()
+          withViewBinding {
+            signInProgressBar.gone()
+            signInEnterButton.enable()
           }
+          Toast.makeText(
+              requireContext(),
+              "Password or Email is wrong!",
+              Toast.LENGTH_SHORT
+          ).show()
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/taskloans/ui/screens/signin/SignInFragment.kt]: you're not checking the error, you're supposing that this is the error

File: app/src/main/java/com/xmartlabs/taskloans/ui/screens/signin/SignInFragment.kt [hunk 3]
```diff
@@ -30,16 +34,18 @@ class SignInFragment : BaseViewBindingFragment<FragmentSigninBinding>() {
...
-                "password or username is wrong, try with userUd = 'xmartlabs', password 'xmartlabs'",
-                Toast.LENGTH_SHORT
-            ).show()
+          withViewBinding {
+            signInProgressBar.gone()
+            signInEnterButton.enable()
           }
+          Toast.makeText(
+              requireContext(),
+              "Password or Email is wrong!",
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/taskloans/ui/screens/signin/SignInFragment.kt]: you should use a string from the string res

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <resources>
     <string name="app_name">TaskLoans</string>
     <string name="screen_sign_in.no_name">noname</string>
-    <string name="user">Usuario</string>
+    <string name="user">Email</string>
```
Comment (mirland) [inline app/src/main/res/values/strings.xml]: change the keys, sign_in.text_hint.user

File: app/src/main/java/com/xmartlabs/taskloans/data/model/service/SignInResponse.kt [hunk 5]
```diff
@@ -1,8 +1,12 @@
...
 import com.xmartlabs.taskloans.data.model.User
 
 /**
  * Created by mirland on 03/05/20.
  */
-data class SignInResponse(val token: String, val user: User)
+data class SignInResponse(
+    @SerializedName("auth_token")
+    val token: String,
+    val user: User)
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/taskloans/data/model/service/SignInResponse.kt]: nl

**Review by mirland — APPROVED** [review]

File: app/src/main/java/com/xmartlabs/taskloans/ui/common/extensions/LiveDataExtensions.kt [hunk 6]
```diff
@@ -7,6 +7,7 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.observe
 import com.xmartlabs.taskloans.device.common.ProcessState
 import com.xmartlabs.swissknife.core.extensions.exhaustive
+import timber.log.Timber
```
Comment (f7deleon) [inline app/src/main/java/com/xmartlabs/taskloans/ui/common/extensions/LiveDataExtensions.kt]: remove this

**Review by f7deleon — APPROVED** [review]

**Pull request merged**